invertase/react-native-firebase
# feat(dynamic-links): add expo config plugin for dynamic-links

### Description
This pull request is intended to add an expo config plugin for @react-native-firebase/dynamic-links which resolves a possible race condition with dynamicLinks().getInitialLink().  The mechanism for this fix is described in [this comment](https://github.com/invertase/react-native-firebase/issues/2660#issuecomment-1271863864).

### Related issues
#2660 
#4548 
https://github.com/invertase/react-native-firebase/issues/4548#issuecomment-1252028059

### Release Summary
feat(dynamic-links): add expo config plugin for dynamic-links
docs(dynamic-links): add note for iOS swizzling workaround

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- My change supports the following platforms;
- My change includes tests;
- This is a breaking change;

### Test Plan

This PR codifies testing that has been done related to [this comment](https://github.com/invertase/react-native-firebase/issues/4548#issuecomment-1252028059) and merely modifies the AppDelegate.  

To test, add the @react-native-firebase/dynamic-links plugin to the list of RNFB plugins, and verify the AppDelegate has been modified.  

Functionally, real world dynamic link matching is inherently hard to test due to the end-to-end nature of the app install process.  My best recommendation for testing here is
* generate a dynamic link that points to your app that is live on the App Store
* return to the Home Screen 
* install a dev version of your app which contains the fix 
* verify getInitialLink is called and contains your expected payload
